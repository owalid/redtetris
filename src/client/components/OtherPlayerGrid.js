import React from 'react';
import _ from 'lodash';
import Game from './Game';

const OtherPlayerGrid = ({ isAlone, mapGamePreview }) => {

  if (!isAlone) {
    const gamePreview = _.cloneDeep(mapGamePreview.currentMapGame)
    gamePreview.map((row, id_row) => {
      row.map((piece, id_piece) => {
        if (piece === 8) {
          gamePreview[id_row][id_piece] = 0
        }
      })
    });
    return (
      <Game
      mapGame={gamePreview}
      isOtherUser={true}
      />
    )
  } 
  return ("")
}

export default OtherPlayerGrid