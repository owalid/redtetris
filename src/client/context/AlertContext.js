import createDataContext from "./createDataContext";
export const ALERT_POP = 'ALERT_POP'

const initialState = {
  message: '',
  type: null,
};

const alertReducer = (state, action) => {
  switch(action.type){
    case ALERT_POP:
      return { message: action.message, type: action.typeAlert }
    default: 
      return state
  }
};

const sendAlert = (dispatch) => (message, typeAlert) =>
  dispatch({ type: ALERT_POP, typeAlert, message })

export const { Provider, Context } = createDataContext(
  //reducer :
  alertReducer,

  //action functions :
  {
    sendAlert
  },

  //initialState :
  {
   ...initialState
  }
);