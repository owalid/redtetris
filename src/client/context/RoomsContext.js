import createDataContext from "./createDataContext";

const initialState = {
  rooms: null
};

const RoomsReducer = (state, payload) => {
  switch (payload.type) {
    case "updateRooms":
      return { ...state, rooms: payload.rooms };
    default:
      return state;
  }
};

const updateRooms = dispatch => rooms => {
  if (Object.keys(rooms).includes("_data")) {
    dispatch({ type: "updateRooms", rooms: rooms._data });
  } else {
    dispatch({ type: "updateRooms", rooms: rooms });
  }
}

export const { Provider, Context } = createDataContext(
  //reducer :
  RoomsReducer,

  //action functions :
  {
    updateRooms
  },

  //initialState :
  {
   ...initialState
  }
);