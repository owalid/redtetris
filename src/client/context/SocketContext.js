import React, { createContext } from "react"
import socketIOClient from "socket.io-client"
require('dotenv').config() 

export const SocketContext = createContext();

export const SocketContextProvider = ({ children }) => {
  const socketClient = socketIOClient(process.env.API_URL);  
  const sendSocket = (type, data = false) => {
    socketClient.emit(type, data);
  }
  
  return (
    <SocketContext.Provider
      value={{ socketClient, sendSocket }}
    >
      { children }
    </SocketContext.Provider>
  )
}