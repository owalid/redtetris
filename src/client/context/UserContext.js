import createDataContext from "./createDataContext";

const initialState = {
  uuidRoom: null,
  player: null
};

const UserReducer = (state, payload) => {
  switch (payload.type) {
    case "updateUuidRoom": {
      return {  ...state, uuidRoom: payload.uuidRoom };
    }
    case "updatePlayer": 
      return {  ...state, player: payload.player };
    default:
      return state;
  }
};

const updateUuidRoom = dispatch => uuidRoom => 
  dispatch({ type: "updateUuidRoom", uuidRoom });

const updatePlayer = dispatch => player => 
  dispatch({ type: "updatePlayer", player });

export const { Provider, Context } = createDataContext(
  //reducer :
  UserReducer,

  //action functions :
  {
    updateUuidRoom,
    updatePlayer
  },

  //initialState :
  {
   ...initialState
  }
);