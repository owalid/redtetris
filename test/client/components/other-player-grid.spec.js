import React, { useContext, useEffect } from "react";
import '@testing-library/jest-dom/extend-expect';
import Enzyme, { mount, shallow } from "enzyme";
import OtherPlayerGrid from '../../../src/client/components/OtherPlayerGrid'
import { render } from '@testing-library/react'
import Adapter from "enzyme-adapter-react-16";
import { player_instance1, rooms_instance, visitor_player, room1 } from '../helpers/data'
import { TestAppSocketProvider } from "../helpers/SocketContext";
import { TestAppUserProvider } from "../helpers/UserContext";
import { TestAppRoomsProvider } from "../helpers/RoomsContext";
import { Context as UserContext } from "../../../src/client/context/UserContext";
import { Context as RoomsContext } from "../../../src/client/context/RoomsContext";

Enzyme.configure({ adapter: new Adapter() });
describe('Alert component', () => {

  const Wrapper = () => (
    <OtherPlayerGrid />
  )

  it('Is exists', () => {
    const wrapper = shallow(<Wrapper />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Can mount', () => {

    const CurrentPlayerSetter = ({ children }) => {
      const { updatePlayer } = useContext(UserContext);
      useEffect(() => {
        updatePlayer(visitor_player);
      }, [])
      return <>{children}</>;
    };

    const CurrentRoomsSetter = ({ children }) => {
      const { updateRooms } = useContext(RoomsContext);
      useEffect(() => {
        updateRooms(rooms_instance);
      }, []);
      return <>{children}</>;
    };

    const Wr = () => (
      <TestAppSocketProvider>
        <TestAppUserProvider>
          <CurrentPlayerSetter>
            <TestAppRoomsProvider>
              <CurrentRoomsSetter>
                <OtherPlayerGrid
                  isAlone={false}
                  mapGamePreview={room1.players[player_instance1.uuid]}
                />
              </CurrentRoomsSetter>
            </TestAppRoomsProvider>
          </CurrentPlayerSetter>
        </TestAppUserProvider>
      </TestAppSocketProvider>
    )
    const w = render(<Wr />);
    expect(w).not.toBeNull()
  })

});