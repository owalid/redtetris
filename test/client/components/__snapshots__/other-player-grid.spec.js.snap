// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component Is exists 1`] = `<OtherPlayerGrid />`;
